anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/util/MetricUtil.java

```diff
@@ -95,4 +98,21 @@ public static MetricData metricData(MetricDefinition metricDef, double value) {
     public static MetricData metricData(MetricDefinition metricDef, double value, long epochSecond) {
         return new MetricData(metricDef, value, epochSecond);
     }
+
+    /**
+     * Convenience method to get metric function if available else return metric key.
+     *
+     * @param mappedMetricData Mapped metric data
+     * @return Returns metric function if available else metric key
+     */
+    public static String getMetricFunctionOrKey(MappedMetricData mappedMetricData) {
+        val metricData = mappedMetricData.getMetricData();
+        val metricDefinition = metricData.getMetricDefinition();
+        val metricTags = metricDefinition.getTags();
+        val functionTagKey = PropertiesUtil.getValueFromProperty(METRIC_FUNCTION_KEY);
+        val metricFunction = metricTags != null && metricTags.getKv() != null
```

Comment:
We should skip this line altogether if `functionTagKey` has not been specified - i.e. it should be optional.